tests/inputFields.spec.ts

```diff
@@ -8,93 +8,61 @@ test.beforeEach( async({page}) => {
+  await pm.onPetTypeEditPage().fillPetType(originalPetType)
+
+  await pm.onPetTypeEditPage().updatePetType()
   
-  /* 9. Add the assertion that the first pet type in the list of names has a value "cat" */
-  // Verifying the input field once again says "cat"
-  await pm.onPetTypesPage().validatePetTypeEntryByRowPosition('cat', 0)
+  await pm.onPetTypesPage().validatePetTypeEntryByIndex(originalPetType, 0)
 
 })
 
 test('Cancel pet type update', async ({page}) => {
+
+  const originalPetType = 'dog'
+  const testPetType = 'moose'
+
   const pm = new PageManager(page)
 
-  /* 1. Select the PET TYPES menu item in the navigation bar */
   await pm.navigateTo().petTypesPage()
 
-  /* 2. Add assertion of the "Pet Types" text displayed above the table with the list of pet types */
-  await pm.onPetTypesPage().validatePageHeader('Pet Types')
-
-  /* 3. Click on "Edit" button for the "dog" pet type */
-  await pm.onPetTypesPage().editPetType('dog')
+  await pm.onPetTypesPage().clickEditPetTypeFor(originalPetType)
+  
+  await pm.onPetTypeEditPage().fillPetType(testPetType)
   
-  /* 4. Type the new pet type name "moose" */
-  /* 5. Add assertion the value "moose" is displayed in the input field of the "Edit Pet Type" page */
-  await pm.onPetTypeEditPage().enterNewPetType('moose')
-
-  /* 6. Click on "Cancel" button */
-  // Clicking the "cancel" button
   await pm.onPetTypeEditPage().cancelPetTypeUpdate()
 
-  /* 7. Add the assertion the value "dog" is still displayed in the list of pet types */
-  // Verifying the inputfield still says "dog"
-  await pm.onPetTypesPage().validatePetTypeEntryByRowPosition('dog', 1)
+  await pm.onPetTypesPage().validatePetTypeEntryByIndex(originalPetType, 1)
 })
 
 test('Pet type name is required validation', async ({page}) => {
   const pm = new PageManager(page)
 
-  /* 1. Select the PET TYPES menu item in the navigation bar */
   await pm.navigateTo().petTypesPage()
 
-  /* 2. Add assertion of the "Pet Types" text displayed above the table with the list of pet types */
-  await pm.onPetTypesPage().validatePageHeader('Pet Types')
-
-  /* 3. Click on "Edit" button for the "lizard" pet type */  
-  /* 4. On the Edit Pet Type page, clear the input field */
-  /* 5. Add the assertion for the "Name is required" message below the input field */
-  await pm.onPetTypesPage().editPetType('lizard')
-  await pm.onPetTypeEditPage().enterNewPetType('')
-  await pm.onPetTypeEditPage().updateNewPetType()
+  await pm.onPetTypesPage().clickEditPetTypeFor('lizard')
 
-  /* 6. Click on "Update" button */
-  // We will try to update the field anyway
-  await pm.onPetTypeEditPage().updateNewPetType()
+  await pm.onPetTypeEditPage().fillPetType('')
 
-  /* 7. Add assertion that "Edit Pet Type" page is still displayed */
-  // Verifying that we are still on the same Edit page
-  await pm.onPetTypeEditPage().validatePageHeader('Edit Pet Type')
+  await pm.onPetTypeEditPage().updatePetType()
+  await pm.onPetTypeEditPage().updatePetType()
```

Comment:
Why do you call method updatePetType() two times?
Also the implementation of this method is confusing, because this method also checks the error message when not message is typed into the input field. It would be better to separate the scope of clicking Update button and validation of error message.

Reply:
Hmmm...  I had a reason for it, but I can no longer remember!  >.< 

I can see how the implementation is not good..  I rearranged the code a little.  Hopefully it is what you meant... 

`   async updatePetType(){
        const petTypeInputField = this.page.getByRole('textbox')

        await this.page.getByText('Update').click()

        if(await petTypeInputField.inputValue() === ''){
            await expect(this.page.locator('div', {hasText: 'Name'}).last()).toHaveText('Name is required')
        }
    }
`

I tested this with two different test cases ('Update pet type', 'Pet type name is required validation').  Both pass - and the second test will cause the error statement validation to run.

Is this what you mean?

Reply:
Actually, I think I got it and am going to go ahead and submit it when ready.  I tested it and it works... confirmed that the 'Name is required' message assertion is picked up.

`async updatePetType(){
      const petTypeInputField = this.page.getByRole('textbox')

      await this.page.getByText('Update').click()

      if(await petTypeInputField.inputValue() === ''){
          await expect(this.page.locator('div', {hasText: 'Name'}).last()).toHaveText('Name is required')
      }
  }`